Automattic/pocket-casts-ios
# Settings Sync: Player Shelf

| 📘 Part of: #1400 |
|:---:|

* Adds synced setting for Player Shelf actions.
* Adds an `Option` type with a `known` and `unknown` case to represent values which we know about and those which we don't. We want to preserve the unknown values to send back to the server.

https://github.com/Automattic/pocket-casts-ios/blob/82d6e59ff96fad321899fc43060156e3df5bc7be/Modules/DataModel/Sources/PocketCastsDataModel/Public/Enums.swift#L64-L67

* The `PlayerAction` enum has been changed to use a `String` `rawValue`, so also include an initializer to convert the old values.

https://github.com/Automattic/pocket-casts-ios/blob/82d6e59ff96fad321899fc43060156e3df5bc7be/Modules/DataModel/Sources/PocketCastsDataModel/Public/Enums.swift#L93-L131

## To test

1. D1: Navigate to the Player
2. D1: Tap the ⋯ icon to rearrange shelf actions
3. D1: Move actions in and out of the top set and around in the list
4. D1: Return to Podcasts list & pull to refresh
5. D2: Pull to refresh Podcasts list
6. D2: Open the player
7. D2: Ensure that shelf actions match

## Checklist
